src/jpegxl.imageio/jxloutput.cpp

```diff
@@ -365,6 +367,111 @@ JxlOutput::write_tiles(int xbegin, int xend, int ybegin, int yend, int zbegin,
 
 
 
+bool
+JxlOutput::save_metadata(ImageSpec& m_spec, JxlEncoderPtr& encoder)
+{
+    DBG std::cout << "JxlOutput::save_metadata()\n";
+
+    // Write EXIF info
+    std::vector<char> exif = { 0, 0, 0, 0 };
+    encode_exif(m_spec, exif);
+
+    // Write XMP packet, if we have anything
+    std::string xmp = encode_xmp(m_spec, true);
+
+    // Write IPTC IIM metadata tags, if we have anything
+    std::vector<char> iptc;
+    encode_iptc_iim(m_spec, iptc);
+
+    bool use_boxes = m_spec.get_int_attribute("jpegxl:use_boxes", 1) == 1;
+    int compress_boxes
+        = m_spec.get_int_attribute("jpegxl:compress_boxes", 0) == 0
+              ? 0
+              : 1;  // TODO: change to 1 for compression by default
```

Comment:
This seems a little harder to follow then necessary (if it equals zero the condition is true, but then in that case set to 0).  Maybe I would suggest the simpler


```suggestion
    int compress_boxes = m_spec.get_int_attribute("jpegxl:compress_boxes") ? 1 : 0;
    // TODO: change to 1 for compression by default
```

Or even change it to bool like the one above?